_pytest/cacheprovider.py

```diff
@@ -14,7 +14,8 @@
 class Cache(object):
     def __init__(self, config):
         self.config = config
-        self._cachedir = config.rootdir.join(".cache")
+        cache_dirname = config.getini("cache_dirname")
+        self._cachedir = config.rootdir.join(cache_dirname)
```

Comment:
to bring this further, people might also use `~` as shortcut form home, and `py.path` doesnt autoresolve those

so correctness unfortunately needs a few more tests and cases handled